docassemble/AssemblyLine/data/questions/al_reminders.yml

```diff
@@ -0,0 +1,89 @@
+---
+mandatory: True
+code: |
+  allow_cron = True
+---
+id: wants reminders
+question: |
+  Would you like to get a reminder about important follow-up steps?
+subquestion: |
+  % if len(al_reminders) == 1:
+  We can send you a reminder, by text or email,
+  about ${ next(iter(al_reminders.values()))["description"][:1].lower() }${ fix_punctuation(next(iter(al_reminders.values()))["description"][1:]) }
+  % else:
+  about:
+
+  % for reminder in al_reminders.values():
+  * ${ reminder["description"]}
+  % endfor
+  % endif
+fields:
+  - I want to get reminders: al_user_wants_reminders
+    datatype: yesnoradio
+  - I want to get reminders by: al_user_preferred_reminder_formats
+    datatype: checkboxes
+    choices:
+      - SMS (text message): sms
+      - Email: email
+    minlength: 1
+    validation messages:
+      minlength: |
+        You need to choose to get reminders by either email or SMS, or both.
+    show if: al_user_wants_reminders    
+  - The email I want to use is: al_user_reminder_email
+    datatype: email
+    default: |
+      % if defined("users[0].email"):
+      ${ users[0].email }
+      % elif user_logged_in():
+      ${ user_info().email }
+      % endif
+    show if: al_user_preferred_reminder_formats["email"]
+  - The phone number I want to use is: al_user_reminder_phone
+    default: |
+      % if defined("users[0].mobile_number"):
+      ${ users[0].mobile_number }
+      % elif defined("users[0].phone_number"):
+      ${ users[0].phone_number }
+      % endif
+    validate: |
+      lambda y: phone_number_is_valid(y) or validation_error("Enter a valid phone number")
+    show if: al_user_preferred_reminder_formats["sms"]
+---
+variable name: al_reminders
+data:
+  filing_documents:
+    description: Filing these documents
+    date: ${ today().plus(days=3).format("yyyy-MM-dd") }
+    email template: al_reminder_filing_template
+    sms template: al_reminder_filing_template
+---
+template: al_reminder_filing_template
+subject: |
+  Did you file your ${ all_variables(special='titles').get('title', AL_ORGANIZATION_TITLE) } documents yet?
+content: |
+  If you haven't filed yet, make a plan or do it today!
+---
+event: cron_daily
+code: |
+  if al_user_wants_reminders:
+    import time
+    from random import random
+    # Avoid all of the interviews using up resources at the same time, vary the start time a bit
+    time.sleep(random())
```

Comment:
Good idea, but is a < 1 second time variation enough? idk how long the email and sms tasks take, maybe more like max 10 or 20 seconds would be good. Guess it also depends on how many of these we end up having.

Reply:
I had ended up fine-tuning this performance on interviews.gbls.org. I had initially run into a big CPU spike when the cron jobs ran on a few hundred interviews. This seemed to be enough variation there, but I agree adding a few extra seconds is a good idea. I did up to 10 minutes originally but lowered it for manual tests.